core/src/wsv.rs

```diff
@@ -943,6 +943,18 @@ impl WorldStateView {
         txs
     }
 
+    /// Get hash of the last transaction in the latest block.
+    #[allow(clippy::expect_used)]
+    pub fn latest_transaction_hash(&self) -> HashOf<VersionedSignedTransaction> {
+        self.latest_block_ref()
+            .expect("WSV should have at least genesis block")
+            .as_v1()
+            .transactions
+            .last()
+            .expect("Block should have at least one transaction")
+            .hash()
+    }
```

Comment:
I think it should either return `Option<...>` or document that it will panic if there are no transactions yet

Reply:
1. `clippy::expect_used` is no longer necessary. 
2. Instead each case of a panic has to be justified. Keep in mind that an `Option<HashOf...` is as efficient memory and instruction wise as panicking, due to niche-value optimisation. 
3. This function should be inline, because a simple case of pointer arithmetic now suddenly comes with two checks and a function call overhead. 
4. I don't quite see why we need this particular function at all. Inlining it into the text of the program might actually make it easier to refactor when `as_v1` isn't the only version.